### PR TITLE
minor changes

### PR DESCRIPTION
**Pull request by inokawazu**
Here are the changes I have made.

- Replaced ⊆ with issubset for definitions and accessibility. 
- Allowed the export `exportqrcode` to export to more types of files by off loading file saving to FILEIO's save.
- Clarified what characters are in the `Byte` mode

### CODE REVIEW

**Comment by RexWzh** [comment]
Hi, @inokawazu. Glad to have you as a new contributor!!

> Replaced ⊆ with issubset for definitions and accessibility.

Not sure why `⊆` should be replaced by `issubset` since the two are referring to the same function. Can you explain more on that?

File: src/QRCoders.jl [hunk 1]
```diff
@@ -196,7 +198,9 @@ function exportqrcode( message::AbstractString
                              mask=mask,
                              compact=compact)
 
-    if !endswith(path, ".png")
+    
+    if !contains("hello", r"\.[a-z0-9]+"i)
```
Comment (johnnychen94) [inline src/QRCoders.jl:202]: Do you mean this?

```suggestion
    if !contains(path, r"\.[a-z0-9]+"i)
```

But I usually check the result of 'splitext'

File: test/tst_overall.jl [hunk 2]
```diff
@@ -1,7 +1,10 @@
-@testset "Exporting a QR code" begin
+@testset "Exporting a QR code to multiple file formats." begin
     message = "To be or not to be a QR code?"
-    exportqrcode(message, imgpath * "qrcode-test")
-    @test true
+    exts = ["", ".png", ".jpg"]
+    for ext in exts
+        exportqrcode(message, imgpath * "qrcode-test" * ext)
```
Comment (johnnychen94) [inline test/tst_overall.jl:5]: There is `@test_nowarn` for this purpose.

**Review by johnnychen94 — APPROVED** [review]
With minor suggestions

File: src/QRCoders.jl (hunk 1 above)
Comment (inokawazu) [reply src/QRCoders.jl:202]: Yes! Thanks for catching that.

**Comment by inokawazu** [comment]
> Hi, @inokawazu. Glad to have you as a new contributor!!
> 
> > Replaced ⊆ with issubset for definitions and accessibility.
> 
> Not sure why `⊆` should be replaced by `issubset` since the two are referring to the same function. Can you explain more on that?

One reason is even though most editors have Unicode support, it is still better IMO to have source code in a widely used character set. Second reason is that ⊆ is not a well known symbol, so for readability and accessibility of the codebase, I thought `issubset` would be better while having no breaking changes. 

> > Allowed the export exportqrcode to export to more types of files by off loading file saving to FILEIO's save.
> 
> This might be considered with #8 in the later works.
> 
> > Clarified what characters are in the Byte mode
> 
> Byte mode does support ISO-8859-1 characters, but it also allows undefined characters range from 0x80 to 0x9f. So I didn't specify the characters set.

Fair point, for sake of clarity I think it would be good to include "but it also allows undefined characters range from 0x80 to 0x9f" into the `Byte()` documentation.

**Comment by RexWzh** [comment]
> One reason is even though most editors have Unicode support, it is still better IMO to have source code in a widely used character set.

You're right. I use `⊆` for math convention, but using `issubset` works well too and might be more convenient in some cases.

> This might be considered with https://github.com/JuliaImages/QRCoders.jl/pull/8 in the later works.

Here I would write something like
```jl
if isnothing(match(r"\.\w+$", path) # no extension
    path *= ".png"
else
    ext = match(r"\.(\w+$)", path).captures[1]
    ext in supportexts || throw("unsupported extension $ext\nsupport list: $supportexts")
end
```

But we can discuss that later. :)

> Fair point, for sake of clarity I think it would be good to include "but it also allows undefined characters range from 0x80 to 0x9f" into the Byte() documentation.

Indeed. I add description about Byte mode in README.md  in https://github.com/JuliaImages/QRCoders.jl/pull/24 as your suggestion.

If you don't mind, could you reset the changes of the middle there commits and add description for the doc of Byte mode. I'll squash these commits and merge into the repo.

Thanks again for these detailed-advice!

**Comment by inokawazu** [comment]
> If you don't mind, could you reset the changes of the middle there commits and add description for the doc of Byte mode. I'll squash these commits and merge into the repo.

@RexWzh, please confirm if this is what you wanted.

**Comment by RexWzh** [comment]
> > If you don't mind, could you reset the changes of the middle there commits and add description for the doc of Byte mode. I'll squash these commits and merge into the repo.
> 
> @RexWzh, please confirm if this is what you wanted.

The original test failed, since the following is not a valid Julia string (more like a python one).
```jl
"xxx\
yyy"
```

**Comment by inokawazu** [comment]
> > > If you don't mind, could you reset the changes of the middle there commits and add description for the doc of Byte mode. I'll squash these commits and merge into the repo.
> > 
> > 
> > @RexWzh, please confirm if this is what you wanted.
> 
> The original test failed, since the following is not a valid Julia string (more like a python one).
> 
> ```julia
> "xxx\
> yyy"
> ```
> 
> For unsupported format, I prefer to use the `EncodeError` to avoid invalid input.

Great, thanks for the corrections. The changes look good to me.

**Pull request merged**